### PR TITLE
Coverity locking

### PR DESCRIPTION
**Pull request by paulidale**
- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/x509/x_x509.c [hunk 1]
```diff
@@ -60,9 +60,12 @@ static int x509_cb(int operation, ASN1_VALUE **pval, const ASN1_ITEM *it,
     case ASN1_OP_NEW_POST:
         ret->ex_cached = 0;
         ret->ex_kusage = 0;
+        if (!CRYPTO_THREAD_write_lock(ret->lock))
```
Comment (slontis) [inline crypto/x509/x_x509.c:63]: Why is the lock just around these 2? What determines this..
It seems like a few others are only set inside ossl_x509v3_cache_extensions which is all inside a lock.

File: crypto/objects/o_names.c [hunk 2]
```diff
@@ -391,9 +391,9 @@ void OBJ_NAME_cleanup(int type)
     if (type < 0) {
         lh_OBJ_NAME_free(names_lh);
         sk_NAME_FUNCS_pop_free(name_funcs_stack, name_funcs_free);
-        CRYPTO_THREAD_lock_free(obj_lock);
         names_lh = NULL;
         name_funcs_stack = NULL;
+        CRYPTO_THREAD_lock_free(obj_lock);
```
Comment (slontis) [inline crypto/objects/o_names.c:396]: Is there still a potential race condition here?

File: crypto/x509/x_x509.c (hunk 1 above)
Comment (paulidale) [reply crypto/x509/x_x509.c:63]: These are the only two that are locked elsewhere, I could must have missed some.  Sledgehammer approach coming up :)

File: crypto/objects/o_names.c (hunk 2 above)
Comment (paulidale) [reply crypto/objects/o_names.c:396]: You mean the `obj_lock = NULL`?

Comment (paulidale) [reply crypto/objects/o_names.c:396]: This whole function is a race condition.  It's deleting stuff without holding the lock.  It's freeing the lock so, I'm guessing, the assumption is that it won't run multi-threaded.

File: crypto/x509/x_x509.c (hunk 1 above)
Comment (slontis) [reply crypto/x509/x_x509.c:61]: Have a look at ossl_x509v3_cache_extensions()..

File: crypto/objects/o_names.c (hunk 2 above)
Comment (slontis) [reply crypto/objects/o_names.c:396]: Yeah I mean what did moving the line down here actually do?

Comment (paulidale) [reply crypto/objects/o_names.c:396]: I think this is best omitted and marked as a false positive.

File: crypto/x509/x_x509.c (hunk 1 above)
Comment (paulidale) [reply crypto/x509/x_x509.c:61]: I moved all of these into the lock.

Comment (t8m) [reply crypto/x509/x_x509.c:61]: Isn't this whole thing just a false positive? I mean this is called only when newly creating an X509 object, is there any point in locking it?

Comment (paulidale) [reply crypto/x509/x_x509.c:61]: It's using a X509 structure that's passed in, one which has a lock already allocated.  It's not clear if it is a false positive or not.

Comment (t8m) [reply crypto/x509/x_x509.c:61]: It is quite clear from what asn1_item_embed_new() does. The lock was initialized right before this callback call so I do not see what would it be protecting against.

Comment (paulidale) [reply crypto/x509/x_x509.c]: I've reverted this file and tagged the two issues as false positivies.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master.  Thanks for the feedback.

**Pull request closed without merging**